packages/list.lua

```diff
@@ -0,0 +1,107 @@
+SILE.scratch.listStack = {}
+
+SILE.settings.declare({
+  name = "list.bullet.numbered",
+  type = "string",
+  default = ".",
+  help = "Bullet used after a list item number"
+})
+
+-- TODO Not ready to handle RTL
+SILE.settings.declare({
+  name = "list.bullet.inset",
+  type = "Length",
+  default = SILE.length.parse("1.3em"),
+  help = "The amount of space by which each level of the list is shifted"
+})
+
+local defaultBullets = {"•", "◦"}
+for i = 1, #defaultBullets do
+  SILE.settings.declare({
+    name = "list.bullet."..i,
+    type = "string",
+    default = defaultBullets[i],
+    help = "Bullet used for unordered list items at level "..i
+  })
+end
+
+SILE.registerCommand("list", function (options, content)
+  local level = #SILE.scratch.listStack + 1
+  local topLevel = level == 1
+  if topLevel then SILE.call("smallskip") else SILE.typesetter:leaveHmode() end
+
+  local numbered = SU.boolean(options.numbered, SU.boolean(options.ordered, options.numbering ~= nil))
+  local bullet = options.bullet
+  if not bullet then
+    if numbered then
+      bullet = SILE.settings.get("list.bullet.numbered")
+    else
+      local setting = "list.bullet."..level
+      -- If the level is too large, cycle bullet types
+      if not SILE.settings.declarations[setting] then
+        local mod = math.mod or math.fmod
+        setting = "list.bullet." .. (1 + mod(level - 1, #defaultBullets))
+      end
+
+      bullet = SILE.settings.get(setting)
+    end
+  end
+  local inset = options.inset and SILE.length.parse(options.inset) or SILE.settings.get("list.bullet.inset")
+
+  SILE.scratch.listStack[level] = {
+    nextNumber = 1,
+    numbered = numbered,
+    numbering = options.numbering or "arabic",
+    bullet = bullet,
+    skip = inset
+  }
+  local baseLSkip = SILE.settings.get("document.lskip")
+  baseLSkip = baseLSkip and baseLSkip.width or SILE.length.new()
+  SILE.settings.temporarily(function ()
+    SILE.settings.set("document.lskip", SILE.nodefactory.newGlue({width = baseLSkip + inset}))
+    SILE.settings.set("document.rskip", options.rskip and SILE.nodefactory.newGlue(options.rskip) or SILE.nodefactory.zeroGlue)
+    SILE.settings.set("document.parindent", SILE.nodefactory.zeroGlue)
+    SILE.process(content)
+  end)
+  SILE.scratch.listStack[#SILE.scratch.listStack] = nil
+
+  if topLevel then SILE.call("smallskip") else SILE.typesetter:leaveHmode() end
+  SILE.call("noindent") -- list does not create a new paragraph
+end)
+
+SILE.registerCommand("list:item", function (options, content)
+  local settings = SILE.scratch.listStack[#SILE.scratch.listStack]
```

Comment:
Using a global. What happens if one has, says, for the mere example a footnote in list, and a list in the footnote? A mess.

Reply:
And what is expected if one mistakenly calls this list:item outside a list.